Instruments/HallProbe_ProLogix.py

```diff
@@ -0,0 +1,101 @@
+import time
+
+class HallProbeController:
+    """
+    Controls a discrete Lake Shore Hall probe via a user-programmable HMCBL-6 cable
+    attached to a Prologix GPIB-USB controller.
+
+    Usage:
+        gpib = Prologix(port='/dev/ttyUSB0', address=5)
+        probe = HallProbeController(gpib)
+        probe.program_user_cable(
+            serial_number=1234567890,
+            current_mA=100,
+            sensitivity_mV_per_kG=1.00
+        )
+        probe.zero_probe()
+        field = probe.read_field()  # returns field in kG
+        Generated with ChatGBT: https://chatgpt.com/share/6851b044-3030-800d-ae80-7943105f6a59
+        Note: method previously named as "_query" is renamed as "_respond".
+    """
+
+    def __init__(self, prologix: Prologix):
+        self.gpib = prologix #Code saves GPIB-USB instance so all methods can call self.gpib.write(...) or self.gpib.read()
+        self.sensitivity = None  # mV/kG, filled after programming
+
+    def _write(self, cmd: str):
+        """Helper to send Prologix commands."""
+        self.gpib.write(f"{cmd}\n") #A write command as you suggested
+    
+    def _respond(self, cmd: str) -> str: #Takes the responses with a write command. Returns the stripped response.
+        """Helper to send a query and return the response."""
+        self._write(cmd)
+        return self.gpib.read().strip()
```

Comment:
these appear to be routines that are designed to ease communication with the prologix, and are not following along with the examples in the modules that I note [above](https://github.com/jmfrancklab/FLinst/pull/38/files#r2159330545).

Reply:
(all that kind of stuff is handled by `gpib_eth`, which you should be inheriting from)